test/new_tests/test_free_threading.py

```diff
@@ -0,0 +1,40 @@
+import threading
+import pytest
+import aerospike
+
+
+@pytest.mark.usefixtures("as_connection")
+class TestFreeThreading:
+    def test_unsafe(self):
+        key = ("test", "demo", 1)
+        BIN_VALUE = 1
+        self.as_connection.put(key, bins={"a": BIN_VALUE})
+
+        bin_value_sum = 0
+
+        THREAD_COUNT = 10
+        barrier = threading.Barrier(parties=THREAD_COUNT)
+        config = {"hosts": [("127.0.0.1", 3000)]}
+        lock = threading.Lock()
+
+        def read_bin():
+            nonlocal barrier, config, lock
+            barrier.wait()
+            client = aerospike.client(config)
+            _, _, bins = client.get(key)
+            nonlocal bin_value_sum
+            lock.acquire()
```

Comment:
You shouldn't have an unnecessary lock here during the read_bin().  You can simply verify by doing something like the following:
1) in the test phase, have your base action be to update a record so the bin value changes from x to x + 1

2) in verification phase, sum the value of all bins you changed before and after the test.  The difference should be the number of operations you submitted.

The key here is that you don't require locking during the submission of the updates